packages/finer-wallet/src/lib/finer-wallet.ts

```diff
@@ -0,0 +1,56 @@
+import {
+  WalletModuleFactory,
+  BrowserWallet,
+  Network,
+} from "@near-wallet-selector/core";
+import {
+  setupMyNearWallet,
+  MyNearWalletParams,
+} from "@near-wallet-selector/my-near-wallet";
+
+export type FinerWalletParams = MyNearWalletParams;
+
+const resolveWalletUrl = (network: Network, walletUrl?: string) => {
+  if (walletUrl) {
+    return walletUrl;
+  }
+
+  switch (network.networkId) {
+    case "mainnet":
+      return "finer://wallet.near.org";
+    case "testnet":
+      return "finer://wallet.testnet.near.org";
+    default:
+      throw new Error("Invalid wallet url");
+  }
+};
+
+export function setupFinerWallet({
+  walletUrl,
+  iconUrl = "./assets/finer-wallet-icon.png",
+  deprecated = false,
+}: FinerWalletParams = {}): WalletModuleFactory<BrowserWallet> {
+  return async (options) => {
+    const wallet = await setupMyNearWallet({
+      walletUrl: resolveWalletUrl(options.options.network, walletUrl),
+      iconUrl,
+    })(options);
+
+    if (!wallet) {
+      return null;
+    }
+
```

Comment:
There should be a check with `is-mobile` to only show Finer Wallet on mobile devices.